### ISSUE TITLE
custom file open dialog

### ISSUE THREAD

**Issue by aexvir** [zed-industries/feedback#288]
**describe the solution you'd like**
not sure how popular my stance is but __I really don't like the native macos file dialog__
I never find my way around it and it takes me much more than what should be needed just to find what I need to open
in the end I mostly resort to invoke whatever tool I need from the terminal 😅 

I got access to the fleet closed beta quite some time ago and one of the features that surprised me by how well they've implemented it is the open file/folder dialog

it's basically just a filesystem tree embedded on it's own modal window

it might not be perfect, but it works really well for opening the right folder/file directly via keyboard
it's closer to the experience of opening files from vim and other terminal based editors, which I personally find really easy to use and agile

**small recording about how it works**
![capture-20220712103126](https://user-images.githubusercontent.com/8055505/178446927-142dc257-c73b-4ca1-b6bd-9018b99f222f.gif)

**Comment by mikayla-maki** [comment]
That reminds me of some file browsers I see on Windows 😂. Speaking for myself, i’m always annoyed when people implement their own file browser because I know how to use the mac file browser so well! But I use the mouse pretty heavily and I see what you mean about keyboard based access.

**Comment by undersquire** [comment]
I think it would be neat to see an option for this that could be toggled.

```json
{
    "use_native_file_dialog": "true",
}
```

Default option could be `false`, so the user opts in for the native file dialog.

**Comment by exalted** [comment]
FWIW, for my very personal needs, OS's default file dialog could be the default.

**Comment by logo4poop** [comment]
I prefer the official apple one myself, but having a second option is good especially for tiled window managers.

**Comment by mikayla-maki** [comment]
This is why we need plan9 😤. Thanks for the research @aexvir, that's a good point!

**Comment by iamnbutler** [comment]
Yeah, with so much ground to cover + the crazy amount of edge/corner cases there would be to tackle a custom file dialog I doubt this would get touched for years. 

Assuming platform support or something in collaboration doesn't lead us to building a custom dialog, I imagine we will use the system one for the foreseeable future.
